src/library_syscall.js

```diff
@@ -1009,6 +1009,8 @@ var SyscallsLibrary = {
     assert(!flags);
 #endif
     if (old.fd === newfd) return -{{{ cDefs.EINVAL }}};
+    // Check newfd is within range of valid open file descriptors.
+    if (newfd < 0 || newfd > 1024) return -{{{ cDefs.EBADF }}};
```

Comment:
Is 1024 a hardcoded limit in musl? Can we perhaps use a `cDefine` for it rather than hardcode it here?

Reply:
Looks like already had FS.MAX_OPEN_FDS